aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1295,15 +1293,6 @@ impl AptosVM {
 
         let (epilogue_session, has_modules_published_to_special_address) = match execution_result {
             Err(execution_error) => {
-                // Invalidate the loader V1 cache in case there was a new module loaded from a
-                // module publish request that failed.
-                // This is redundant with the logic in execute_user_transaction but unfortunately is
-                // necessary here as executing the underlying call can fail without this function
-                // returning an error to execute_user_transaction.
-                if *new_published_modules_loaded && !self.features().is_loader_v2_enabled() {
```

Comment:
Getting rid of these is such a milestone.
tagging @zekun000 as the chief code quality appreciation officer :)